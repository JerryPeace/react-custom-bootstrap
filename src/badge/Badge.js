import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import bootstrapUtils from '../utils/bootstrapUtils';

class Badge extends React.Component {
    static propTypes = {
        bsClass: PropTypes.string
    };

    static defaultProps = {
        bsClass: 'badge'
    };

    render() {
        const {bsClass, ...rest} = this.props;
        let bsCls = bootstrapUtils.getCls(this.props);

        return (
            <span {...rest} className={classNames(this.props.className, bsCls)}>
                {this.props.children}
            </span>
        );
    }
}

export default Badge;
