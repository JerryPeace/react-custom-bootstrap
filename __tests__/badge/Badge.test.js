import React from 'react';
import { mount } from 'enzyme';
import toJSON from'enzyme-to-json';
import Badge from 'badge/Badge';

describe('Badge', () => {
    test('render badge default', () => {
        const form = mount(
            <Badge>
                <div>test</div>
            </Badge>
        )
        expect(toJSON(form)).toMatchSnapshot();
    });
    test('render badge with props', () => {
        const form = mount(
            <Badge className='test_class' prop1='prop1' prop2='prop2'>
                <div>test</div>
            </Badge>
        )
        expect(toJSON(form)).toMatchSnapshot();
    });
});
