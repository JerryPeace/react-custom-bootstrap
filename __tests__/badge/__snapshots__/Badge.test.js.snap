// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge render badge default 1`] = `
<Badge
  bsClass="badge"
>
  <span
    className="badge"
  >
    <div>
      test
    </div>
  </span>
</Badge>
`;

exports[`Badge render badge with props 1`] = `
<Badge
  bsClass="badge"
  className="test_class"
  prop1="prop1"
  prop2="prop2"
>
  <span
    className="test_class badge"
    prop1="prop1"
    prop2="prop2"
  >
    <div>
      test
    </div>
  </span>
</Badge>
`;
